presto-test-jdbc-compatibility-old-server/pom.xml

```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.prestosql</groupId>
+        <artifactId>presto-root</artifactId>
+        <version>341-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>presto-test-jdbc-compatibility-old-server</artifactId>
+    <description>Presto - Tests whether current Presto JDBC clients are compatible with old Presto servers</description>
+
+    <properties>
+        <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+        <dep.presto-jdbc-under-test>341-SNAPSHOT</dep.presto-jdbc-under-test>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.prestosql</groupId>
+            <artifactId>presto-jdbc</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>io.prestosql</groupId>
+            <artifactId>presto-jdbc</artifactId>
+            <scope>test</scope>
+            <version>${project.version}</version>
```

Comment:
I don't think we should run the tests in old versions of the project. The "source of truth" for tests should be what's in trunk right now. The problems with running old test code are:
* If there are bug in the old tests, there's no way to "fix" them -- we can't go back in time and patch an old presto test jar
* It relies on treating the structure of the tests as an API (it's not!). If we refactor the test code in master, the old tests may stop running at all.

If we want more test coverage, or tests targeted to a specific version, we should augment the *current* test library and enable tests conditionally based on the version being targeted.

Reply:
> I don't think we should run the tests in old versions of the project. The "source of truth" for tests should be what's in trunk right now

This code does not run old tests. The version of tests which is run is the one living on the trunk. What this PR does is running those tests against older versions of Presto servers run in docker.
I even had to disable a few test assertions, for queries which did not work before parametric timestamp was introduced (see: https://github.com/prestosql/presto/pull/4998/commits/f3449b32d5831a37780dc27c5a067d9d586a8148)

(The `<version>${project.version}</version>` can be dropped as this is the default defined in parent pom.)